R/utilities-observed-data.R

```diff
@@ -407,3 +504,30 @@ getObservedDataIdFromPath <- function(path) {
   pathArray <- ospsuite::toPathArray(path)
   return(pathArray[1])
 }
+
+#' @title translateDataSelection
+#' @description
+#' Translate `dataSelection` input by user into characters/expression understood by `getSelectedData`
+#' @param dataSelection characters or expression to select subset the observed data
+#' @return characters or expression to select subset the observed data
+#' @keywords internal
+translateDataSelection <- function(dataSelection){
+  validateIsOfType(dataSelection, c("character", "expression"), nullAllowed = TRUE)
+  if (!isOfType(dataSelection, "character")) {
+    return(dataSelection)
+  }
+  # If any selection include None, do not select anything
+  if (isIncluded(DataSelectionKeys$NONE, dataSelection)) {
+    return(FALSE)
+  }
+  # By removing "" string from dataSelection
+  # If "" is the only value provided, dataSelection isEmpty
+  # If multiple values provided, concatenate the remaining selections
+  dataSelection <- dataSelection[!(dataSelection %in% "")]
```

Comment:
before that, dataSelection should be trimmed, e.g. with str_trim() from stringr:
```
> dataSelection=c(" aa ", " bb ", " ")
> dataSelection=str_trim(dataSelection)
> dataSelection
[1] "aa" "bb" ""  
```

Reply:
After the current line 526 we should also add the check `if(isEmpty(dataSelection)) return(FALSE)`
Otherwise the function returns "()" which is not really convenient and also not consistent with the FALSE return in case of `DataSelectionKeys$NONE`
(and then you could also remove checking for "()" when calling the function ;)